src/txn/txn_ckpt.c

```diff
@@ -1271,6 +1271,93 @@ __drop_to(WT_CKPT *ckptbase, const char *name, size_t len)
 	}
 }
 
+/*
+ * __checkpoint_lock_dirty_tree_requires_hot_backup_lock --
```

Comment:
This naming is rather verbose. It really is the main content of the original function, I just call it `__checkpoint_lock_dirty_tree_int` as the internal version of the function with the lock.

Reply:
bc1914f